AprilSylph/XKit-Rewritten
# feat(NotificationBlock): Add optional placeholders for hidden notifications

### Description


Adds placeholder text to the activity page in places where at least one notification is blocked. This is done a bit intelligently; a group of sequential blocked notifications on the same day are replaced with a single placeholder.

Not sure if I would make this on-by-default or off-by-default. 

<img width="575" src="https://github.com/user-attachments/assets/0b54b468-03df-45f5-92d5-b5ae6781193e">

### Testing steps


- Confirm that blocking/unblocking notifications on a post causes relevant notifications to appear/disappear without having to refresh the page.

Enable the new preference.

- Block notifications for a post with multiple notifications in a row. Confirm that a placeholder element replaces the blocked notifications.
- Block notifications for a post with multiple notifications in a row, spanning multiple days. Confirm that a placeholder element replaces the blocked notifications for each day, resulting in the day labels still making sense.

Feel free to express an opinion about the default status of this new preference.
